include/xwidgets/xtransport.hpp

```diff
@@ -315,18 +321,29 @@ namespace xw
     template <class P>
     inline void xtransport<D>::notify(const P& property) const
     {
+        xeus::xjson state;
+        xeus::buffer_sequence buffers;
+        set_patch_from_property(property, state, buffers);
+
         if (m_hold != nullptr)
         {
-            auto it = m_hold->find(property.name());
-            if (it != m_hold->end() && it.value() == property())
```

Comment:
`it.value() == property()` was causing a JSON serialization of the property.

We need to compare with the full serialization.